### DISCUSSION TITLE
Global file/part history or successcounter

### DISCUSSION THREAD

**Discussion by JFSolution-old** [OctoFarm/OctoFarm#210] [Feature Request]
It would be cool to see how often a specifiy file was successfully printed, in total for all printers.

And a further step would be to manage this on Part level.
So if you have a file with five copys of a part, and one with 10 copys.
You say the system the amount of the part in the file and on a successfull print, the successcounter of the Part get increased by the amount.

**Comment by NotExpectedYet** [comment]
I like the idea, does this not work for you currently though? or maybe I'm mis-understanding. 

![image](https://user-images.githubusercontent.com/61137988/103649411-51050f80-4f56-11eb-9aaf-37334ac35381.png)

Just select the file name on History and it should give you the status across all printers. 

I would however like to make that easier which I'm working on for the UI changes going forward. Currently fails for anyone using printer names inside the file name as well as those would be different across each printer. 

Would you mind explaining this a bit more please, I didn't understand unless your just refering to a live counter on the page?  
>You say the system the amount of the part in the file and on a successfull print, the successcounter of the Part get increased by the >amount.

**Reply by JFSolution-old** [reply]
My idea was a new Model, for example called Part.
Here you can set up a Part and choose which file across all printers contains this part and which amount of the part.
And in the filemanager you see a new column for this counter.
So you have a overview of the amount of this Part, if you want to start this print.
In the Part view you can reset the counter.

a example use case:
Someone orders 15 Faceshields from you and you print this just on demand.
You have files for every of your printer, with different filenames with different amount of this faceshields.
So you want to know how many of this you already printed if you start a new one.

As I‘m writing this down, i get in mind, that you would need a value where you can set the needed amount. And in the fileexplorer you have to see the amount of successfull printed, in progress and needed parts.

I would like to help with this, but I don‘t have any expirience with nodejs and need some time to learn this.

**Reply by NotExpectedYet** [reply]
I like the new model idea certainly. 

I think this would fit perfectly into my Job/Queues implementation that's planned. I would need that first to generate the required information to then be able to show it in the file manager as you wanted. I do see the utilisation of including that information there but I would like to tackle the global file manager / Job Queues first because then all of this information would be readily available to call from an API. 

If you do eventually learn node then please message me as there is a lot of clean up coming on the code side of things. Currently I'm not accepting PR requests due to this, basically me been lazy with code, I wouldn't force understanding this on anyone! After those changes are complete it will be in a more understandable state and we can work in sync as well.

**Reply by NotExpectedYet** [reply]
I may merge this with #219 once I figure out how haha. 

Thanks for the suggestions too, they are good.

**Reply by NotExpectedYet** [reply]
This should now available in history on v1.2

**Reply by NotExpectedYet** [reply]
Well kind of it displays the file name not part.
